### PR TITLE
Public Release for Fall24 meta-release

### LINKED ISSUE #168: Remove the reference to a `selected protocol` inside the `sink`-description
**Problem description**
For sure based on Commonalities you normally can also use a `protocol`-property.
This property is not used in CarrierBilling (for now).
So I guess this reference can be removed.

```yaml
        sink:
          type: string
          format: url
          description: The address to which events shall be delivered using the selected protocol.
          example: "https://endpoint.example.com/sink"
```

**Expected behavior**

```yaml
        sink:
          type: string
          format: url
          description: The address to which events shall be delivered.
          example: "https://endpoint.example.com/sink"
```

### LINKED ISSUE #173: Should the Generic401 code value be "UNAUTHENTICATED" instead of "UNAUTHORIZED"
**Problem description**
In the [CAMARA_common.yaml](https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/artifacts/CAMARA_common.yaml#L173), the 401 code is `UNAUTHENTICATED`:
```
            GENERIC_401_UNAUTHENTICATED:
              description: Request cannot be authenticated
              value:
                status: 401
                code: UNAUTHENTICATED
                message: Request not authenticated due to missing, invalid, or expired credentials.
```

However in [carrier_billing.yaml](https://github.com/camaraproject/CarrierBillingCheckOut/blob/main/code/API_definitions/carrier_billing.yaml#L1596) and [carrier_billing_refund.yaml](https://github.com/camaraproject/CarrierBillingCheckOut/blob/main/code/API_definitions/carrier_billing_refund.yaml#L1033), the code is `UNAUTHORIZED`:
```
          example:
            code: UNAUTHORIZED
            status: 401
            message: "Authorization failed: ..."
```

Should the value be aligned with [CAMARA_common.yaml](https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/artifacts/CAMARA_common.yaml#L173)? I believe many other CAMARA APIs are also using `UNAUTHENTICATED`.

**Expected behavior**
The `code` value to be changed in [carrier_billing.yaml](https://github.com/camaraproject/CarrierBillingCheckOut/blob/main/code/API_definitions/carrier_billing.yaml#L1596) and [carrier_billing_refund.yaml](https://github.com/camaraproject/CarrierBillingCheckOut/blob/main/code/API_definitions/carrier_billing_refund.yaml#L1033) to align with [CAMARA_common.yaml](https://github.com/camaraproject/Commonalities/blob/r0.4.0-rc.2/artifacts/CAMARA_common.yaml#L173)

### LINKED ISSUE #175: Convention for file names of API definition
**Problem description**
There is currently no guidelines for file names of API definition files in CAMARA.
The initially agreed proposal is to use the convention api-name.yaml as a guideline for the API definition file names.

**Expected behavior**
Change the file names to kebab-case before or as a part of release PR for M4.

**Alternative solution**
Wait for the final guidelines.

**Additional context**
See the discussion in [Commonalities](https://github.com/camaraproject/Commonalities/issues/284)

### PR DESCRIPTION
**Pull request by PedroDiez**
#### What type of PR is this?

Add one of the following kinds:
* bug
* correction
* documentation
* subproject management

#### What this PR does / why we need it:

PR to generate Public Release of Carrier Billing Checkout Family APIs

#### Which issue(s) this PR fixes:

Fixex #155 
Fixes #168
Fixes #173
Fixes #175 

#### Special notes for reviewers:

**PRs #174 and #176 to be merged first than this one**

Changes driven in API Spec YAMLs:
- Apply File Name conventions (kebab-case)
- Adjustment of version and servers values
- Adjustment externalDocs link
- Adjustment sink Description
- Fix missing required request Body in `confirmPayment` operation
- Fix 401 Error with Commonalities and also align the model of Errors with Commonalities "CAMARA_common.yaml" naming. - - Remove 400 PhoneNumber Required as Covered by 422 Exception

#### Changelog input

```
Public Release of Carrier Billing Family APIs for MetaRelease Fall24

```

#### Additional documentation 

N/A

### CODE REVIEW

**Review by bigludo7 — DISMISSED** [review]
LGTM - Thanks @PedroDiez 
Up to @hdamker and the release Management team ;)

**Comment by hdamker** [comment]
@PedroDiez If you want to have the meeting soon within main, you may want to do it in an own PR. Could take some days before the release PR gets approved by the RM team.

**Review by jlurien — CHANGES_REQUESTED** [review]
- API Readiness checklists should be updated to r1.2 versions (without -rc suffixes), and resolve any pending tbd

**Comment by PedroDiez** [comment]
> @PedroDiez If you want to have the meeting soon within main, you may want to do it in an own PR. Could take some days before the release PR gets approved by the RM team.

Thanks Herbert, I will comment tomorrow in the WG meeting. I think we can wait because relevant people is already aware of new schedule

File: documentation/API_documentation/carrier-billing-refund-API-Readiness-Checklist.md
Comment (jlurien) [inline documentation/API_documentation/carrier-billing-refund-API-Readiness-Checklist.md:12]:
```suggestion
|  6 | User stories                                 |   O   |         O         |    O    |    M   |  Y   | [link](/documentation/API_documentation/Carrier Billing Refund User Story.md) |
```

**Review by jlurien — COMMENTED** [review]
Some projects are naming User stories with the same format than API-Readiness-Checklist, that is api-name-User-Story.md, but it is not made a requirement

**Comment by PedroDiez** [comment]
> Some projects are naming User stories with the same format than API-Readiness-Checklist, that is api-name-User-Story.md, but it is not made a requirement

Thanks @jlurien I am applying it, It is fine to align

**Review by jlurien — APPROVED** [review]
Approved on behalf of Release Management

**Comment by PedroDiez** [comment]
After TSC meeting, it can be merged:

SC Approval of M4 (all release PRs of Sub Project reviewed by Release Management)
- We can merge PR when review completed
- M4 Release tracker date:  date when the merged is done by the project

**Pull request merged**
